vectorsearch/resource_vector_search_endpoint.go

```diff
@@ -0,0 +1,85 @@
+package vectorsearch
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/databricks/databricks-sdk-go/service/vectorsearch"
+)
+
+const DefaultProvisionTimeout = 45 * time.Minute
+
+func ResourceVectorSearchEndpoint() *schema.Resource {
+	s := common.StructToSchema(
+		vectorsearch.EndpointInfo{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(s, "name").SetRequired()
+			s["name"].ForceNew = true
+			s["endpoint_type"].ForceNew = true
+			common.CustomizeSchemaPath(s, "endpoint_type").SetRequired()
+			delete(s, "id")
+			common.CustomizeSchemaPath(s, "creator").SetReadOnly()
+			common.CustomizeSchemaPath(s, "creation_timestamp").SetReadOnly()
+			common.CustomizeSchemaPath(s, "last_updated_timestamp").SetReadOnly()
+			common.CustomizeSchemaPath(s, "last_updated_user").SetReadOnly()
+			common.CustomizeSchemaPath(s, "endpoint_status").SetReadOnly()
+			common.CustomizeSchemaPath(s, "num_indexes").SetReadOnly()
+			common.CustomizeSchemaPath(s).AddNewField("endpoint_id", &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			})
```

Comment:
This is identical to the `id`, right? Why do we need a separate field for this?

Reply:
No. The `id` is the name... This is some random ID that isn't used in the request

Reply:
I see... let me check with them. It doesn't seem to even be used in any other part of the REST API. Why is it even exposed?

Reply:
It could be used for some internal tracking, like, permissions when they will be added... I had the same situation with Model Serving.

Reply:
yes. Here is the same for the model serving: https://github.com/databricks/terraform-provider-databricks/blob/main/serving/resource_model_serving.go#L69